http/response.go

```diff
@@ -74,34 +85,27 @@ func (res *Response) RawNext() (interface{}, error) {
 	}
 	m := &cmds.MaybeError{Value: value}
 	err := res.dec.Decode(m)
-
-	// last error was sent as value, now we get the same error from the headers. ignore and EOF!
-	if err != nil && res.err != nil && err.Error() == res.err.Error() {
-		err = io.EOF
-	}
-
-	return m.Get(), err
-}
-
-func (res *Response) Next() (interface{}, error) {
-	v, err := res.RawNext()
 	if err != nil {
-		return nil, err
+		if err == io.EOF {
+			res.err = err
```

Comment:
I may be missing something, but do we need to check for trailing error headers here?

Reply:
Yes, you're totally right. Building a comprehensive test suite for cmds would be a big and important project by itself...